pkg/useragent/useragent.go

```diff
@@ -0,0 +1,61 @@
+// package useragent provides a consistent way to get a user agent for outbound http requests from Grafana Agent.
+// The default User-Agent is `GrafanaAgent/$VERSION($MODE)`
+// Where version is the build version of the agent and MODE is one of "static" or "flow".
+package useragent
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/grafana/agent/pkg/build"
+)
+
+const (
+	operatorEnv = "AGENT_OPERATOR"
+	modeEnv     = "AGENT_MODE"
+)
+
+// settable by tests
+var goos = runtime.GOOS
+
+func UserAgent() string {
```

Comment:
Calling this `Get()` might be less repetitive so callers use `useragent.Get()` instead of `useragent.UserAgent()`